test/lightning_web/plugs/webhook_auth_test.exs

```diff
@@ -7,15 +7,89 @@ defmodule LightningWeb.Plugs.WebhookAuthTest do
   alias LightningWeb.Plugs.WebhookAuth
   alias Lightning.Repo
 
+  @moduletag capture_log: true
+
   setup do
     {:ok, trigger: insert(:trigger), auth_method: insert(:webhook_auth_method)}
   end
 
+  test "OPTIONS preflight is a no-op" do
+    conn = conn(:options, "/i/whatever") |> WebhookAuth.call([])
+    refute conn.halted
+    assert conn.status == nil
+    assert conn.assigns[:trigger] == nil
+  end
+
+  test "non-/i path passes through unchanged" do
+    conn = conn(:post, "/api/things") |> WebhookAuth.call([])
+    refute conn.halted
+    assert conn.status == nil
+    assert conn.assigns[:trigger] == nil
+  end
+
+  test "responds 404 for wrong x-api-key on protected trigger", %{
+    trigger: trigger
+  } do
+    api_method = insert(:webhook_auth_method, auth_type: :api, api_key: "secret")
+    associate_auth_method(trigger, api_method)
+
+    conn =
+      conn(:post, "/i/#{trigger.id}")
+      |> put_req_header("x-api-key", "nope")
+      |> WebhookAuth.call([])
+
+    assert conn.halted
+    assert conn.status == 404
+    assert Jason.decode!(conn.resp_body) == %{"error" => "webhook_not_found"}
+  end
+
+  test "assigns trigger for matching x-api-key", %{trigger: trigger} do
+    api_method = insert(:webhook_auth_method, auth_type: :api, api_key: "secret")
+    associate_auth_method(trigger, api_method)
+
+    conn =
+      conn(:post, "/i/#{trigger.id}")
+      |> put_req_header("x-api-key", "secret")
+      |> WebhookAuth.call([])
+
+    expected_trigger =
+      trigger
+      |> unload_relation(:workflow)
+      |> Repo.preload([:workflow, :edges, :webhook_auth_methods])
+
+    refute conn.halted
+    assert conn.assigns[:trigger] == expected_trigger
```

Comment:
In cases where I am testing code that is performing a lookup, I like to have "negative" examples if it is cheap to do so - e.g. at least one other instance of trigger (in this case) that I am not interested to try and offer some reassurance that my code will still work when there is more than one instance of the thing that I am looking for (i.e. the way things would be in a non-test env).